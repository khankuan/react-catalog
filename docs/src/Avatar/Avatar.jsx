import React from 'react'
import PropTypes from 'prop-types';
import cx from 'classnames'

import './Avatar.css'

/**
 *  @categories: Image
 *  @tags: Profile, Picture
 *  @default
 *  Display pic of a user
 */
const Avatar = ({ theme, type, src, ...passProps }) => {
  return (
    <img
      {...passProps}
      className={cx('demo-avatar', `demo-avatar-type-${type}`, `demo-avatar-theme-${theme}`)}
      src={src} />
  )
}

Avatar.propTypes = {
  /** Image Url */
  src: PropTypes.string.isRequired,
  type: PropTypes.oneOf(['square', 'round']),
  theme: PropTypes.oneOf(['light', 'dark'])
}

Avatar.defaultProps = {
  src: 'http://success.grownupgeek.com/wp-content/uploads/2013/01/no-avatar.png',
  type: 'round',
  theme: 'light'
}

export default Avatar
