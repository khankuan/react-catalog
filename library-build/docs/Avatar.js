export default {
  "description": "Display pic of a user",
  "methods": [],
  "dependencies": [
    "img"
  ],
  "props": {
    "src": {
      "type": {
        "name": "string"
      },
      "required": true,
      "description": "Image Url",
      "defaultValue": {
        "value": "'http://success.grownupgeek.com/wp-content/uploads/2013/01/no-avatar.png'",
        "computed": false
      }
    },
    "type": {
      "type": {
        "name": "enum",
        "value": [
          {
            "value": "'square'",
            "computed": false
          },
          {
            "value": "'round'",
            "computed": false
          }
        ]
      },
      "required": false,
      "description": "",
      "defaultValue": {
        "value": "'round'",
        "computed": false
      }
    },
    "theme": {
      "type": {
        "name": "enum",
        "value": [
          {
            "value": "'light'",
            "computed": false
          },
          {
            "value": "'dark'",
            "computed": false
          }
        ]
      },
      "required": false,
      "description": "",
      "defaultValue": {
        "value": "'light'",
        "computed": false
      }
    }
  },
  "name": "Avatar",
  "categories": [
    "Image"
  ],
  "tags": [
    "Profile",
    "Picture"
  ],
  "hasDefault": true
}