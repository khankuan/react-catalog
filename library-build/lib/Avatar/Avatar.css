.demo-avatar {
  height: 40px;
  width: 40px;
  display: inline-block;
  border-width: 3px;
  border-style: solid;
}

.demo-avatar-type-round {
  border-radius: 22px;
}

.demo-avatar-type-square {
  border-radius: 2px;
}

.demo-avatar-theme-light {
  border-color: #ccc;
}

.demo-avatar-theme-dark {
  border-color: #999;
}
