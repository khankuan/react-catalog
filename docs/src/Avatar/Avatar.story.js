import { Story } from 'react-library'
import Avatar from './Avatar'

const story = new Story(Avatar)
export default story

story.add({
  title: 'Square type',
  content: { type: 'square' }
})

story.add({
  title: 'Dark theme',
  content: { theme: 'dark' }
})
