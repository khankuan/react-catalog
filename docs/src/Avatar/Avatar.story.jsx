import { Story } from 'react-catalog'
import Avatar from './Avatar'

const story = new Story(Avatar)
export default story

story.add({
  title: 'Square type',
  content: { type: 'square' }
})

story.add({
  title: 'Dark theme',
  content: { theme: 'dark' }
})
