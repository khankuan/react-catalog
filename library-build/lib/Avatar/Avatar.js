'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

require('./Avatar.css');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 *  @categories: Image
 *  @tags: Profile, Picture
 *  @default
 *  Display pic of a user
 */
var Avatar = function Avatar(_ref) {
  var theme = _ref.theme,
      type = _ref.type,
      src = _ref.src,
      passProps = (0, _objectWithoutProperties3.default)(_ref, ['theme', 'type', 'src']);

  return _react2.default.createElement('img', (0, _extends3.default)({}, passProps, {
    className: (0, _classnames2.default)('demo-avatar', 'demo-avatar-type-' + type, 'demo-avatar-theme-' + theme),
    src: src }));
};

Avatar.propTypes = {
  /** Image Url */
  src: _react.PropTypes.string.isRequired,
  type: _react.PropTypes.oneOf(['square', 'round']),
  theme: _react.PropTypes.oneOf(['light', 'dark'])
};

Avatar.defaultProps = {
  src: 'http://success.grownupgeek.com/wp-content/uploads/2013/01/no-avatar.png',
  type: 'round',
  theme: 'light'
};

exports.default = Avatar;