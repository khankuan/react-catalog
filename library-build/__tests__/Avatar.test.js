
import React from 'react'
import Avatar from '../../docs/src/Avatar/Avatar.jsx'
import ReactTestUtils from 'react-addons-test-utils'
it('Avatar default renders', () => {
  expect(ReactTestUtils.createRenderer().render(<Avatar />)).toMatchSnapshot();
})
  