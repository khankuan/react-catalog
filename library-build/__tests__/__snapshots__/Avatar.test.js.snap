exports[`test Avatar default renders 1`] = `
<img
  className="demo-avatar demo-avatar-type-round demo-avatar-theme-light"
  src="http://success.grownupgeek.com/wp-content/uploads/2013/01/no-avatar.png" />
`;
